File: api/src/main/java/org/pmiops/workbench/cohorts/CohortFactory.java
package org.pmiops.workbench.cohorts;

import org.pmiops.workbench.db.model.DbCohort;
import org.pmiops.workbench.db.model.CohortReview;
import org.pmiops.workbench.db.model.DbWorkspace;
import org.pmiops.workbench.db.model.User;

public interface CohortFactory {

  DbCohort createCohort(org.pmiops.workbench.model.Cohort apiCohort, User creator, long workspaceId);

  DbCohort duplicateCohort(String newName, User creator, DbWorkspace targetWorkspace, DbCohort original);

  CohortReview duplicateCohortReview(CohortReview original, DbCohort targetCohort);
}

File: api/src/main/java/org/pmiops/workbench/cohorts/CohortFactory.kt
package org.pmiops.workbench.cohorts

import org.pmiops.workbench.db.model.Cohort
import org.pmiops.workbench.db.model.CohortReview
import org.pmiops.workbench.db.model.User
import org.pmiops.workbench.db.model.Workspace

interface CohortFactory {

    fun createCohort(apiCohort: org.pmiops.workbench.model.Cohort, creator: User, workspaceId: Long): Cohort

    fun duplicateCohort(newName: String, creator: User, targetWorkspace: Workspace, original: Cohort): Cohort

    fun duplicateCohortReview(original: CohortReview, targetCohort: Cohort): CohortReview
}
